src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs

```diff
@@ -135,7 +135,10 @@ public ICollection<ModelState> Values
         /// <inheritdoc />
         public bool IsValid
         {
-            get { return ValidationState == ModelValidationState.Valid; }
+            get
+            {
+                return ErrorCount == 0;
+            }
```

Comment:
> For scenarios involving TryUpdateModel this is broken for now

This might be a big issue since it breaks Mvc 5 expectation. Should TryUpdateModel do validation for you as a post step?